File: BinaryTreePaths/BinaryTreePaths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        res, stack = [], [(root, "")]
        while stack:
            node, ls = stack.pop()
            if not node.right and not node.left:
                res.append(ls + str(node.val))
            if node.right:
                stack.append((node.right, ls + str(node.val) + "->" ))
            if node.left:
                stack.append((node.left, ls + str(node.val) + "->"))
        return res


class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        if not root:
            return []
        return [str(root.val) + '->' + path
                for kid in (root.left, root.right) if kid
                for path in self.binaryTreePaths(kid)] or [str(root.val)]

File: BinaryTreePaths/BinaryTreePaths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function (root) {
  if (!root) return [];
  let result = [];
  function path(root, str) {
    if (!root.left && !root.right) result.push(str + root.val);
    if (root.left) path(root.left, str + root.val + "->");
    if (root.right) path(root.right, str + root.val + "->");
  }
  path(root, "");
  return result;
};
